common/metrics/defs.go

```diff
@@ -1070,21 +1072,22 @@ var metricDefs = map[ServiceIdx]map[int]metricDefinition{
 
 	// definitions for Replicator metrics
 	Replicator: {
-		ReplicatorWsFailure:                             {Counter, "replicator.websocket.failure"},
+		ReplicatorCreateInStreamFailure:                 {Counter, "replicator.create-in-stream.failure"},
+		ReplicatorCreateOutStreamFailure:                {Counter, "replicator.create-out-stream.failure"},
 		ReplicatorRequests:                              {Counter, "replicator.requests"},
 		ReplicatorFailures:                              {Counter, "replicator.errors"},
 		ReplicatorBadRequest:                            {Counter, "replicator.requests.bad"},
 		ReplicatorInConnCreditsReceived:                 {Counter, "replicator.inconn.creditsreceived"},
 		ReplicatorInConnMsgWritten:                      {Counter, "replicator.inconn.msgwritten"},
 		ReplicatorOutConnCreditsSent:                    {Counter, "replicator.outconn.creditssent"},
 		ReplicatorOutConnMsgRead:                        {Counter, "replicator.outconn.msgread"},
-		ReplicatorReconcileDestRun:                      {Counter, "replicator.reconcile.dest.run"},
-		ReplicatorReconcileDestFail:                     {Counter, "replicator.reconcile.dest.fail"},
-		ReplicatorReconcileDestFoundMissing:             {Counter, "replicator.reconcile.dest.foundmissing"},
-		ReplicatorReconcileDestExtentRun:                {Counter, "replicator.reconcile.destextent.run"},
-		ReplicatorReconcileDestExtentFail:               {Counter, "replicator.reconcile.destextent.fail"},
-		ReplicatorReconcileDestExtentFoundMissing:       {Counter, "replicator.reconcile.destextent.foundmissing"},
-		ReplicatorReconcileDestExtentInconsistentStatus: {Counter, "replicator.reconcile.destextent.inconsistentstatus"},
+		ReplicatorReconcileDestRun:                      {Gauge, "replicator.reconcile.dest.run"},
```

Comment:
Don't you want to also update the callers to use UpdateGauge() instead of IncCounter() now ?